### PR TITLE
Feature/add start and end time for trip routes

### PR DESCRIPTION
**Pull request by ruig1992**
[trs Saving start and end times for each of the trip routes](https://trello.com/c/skJK6zgT)

---

**TODO in the future** :: Need to change the some code to work with the collection of **Waypoints** instances and the collection of **Routes** instances.

`\App\Http\Requests\CreateTripRequest`, `\App\Http\Requests\UpdateTripRequest`
`\App\Services\TripsService` - `getRoutesFromWaypoints()`, `getRoutesTimeForReverseTrip()`

### CODE REVIEW

**Review by TatsuUkraine — APPROVED** [review]

**Comment by ruig1992** [comment]
@TatsuUkraine  Conflicts resolved.

File: resources/assets/js/features/trip/components/Containers/CreateTripContainer.js [hunk 1]
```diff
@@ -39,6 +42,18 @@ class CreateTripContainer extends React.Component {
                 place: null,
             }
         };
+
+        this.tripEndTime = 0;
```
Comment (TatsuUkraine) [inline resources/assets/js/features/trip/components/Containers/CreateTripContainer.js]: use local state

File: resources/assets/js/features/trip/components/Containers/CreateTripContainer.js [hunk 2]
```diff
@@ -39,6 +42,18 @@ class CreateTripContainer extends React.Component {
...
+        this.waypointsDurations = [];
```
Comment (TatsuUkraine) [inline resources/assets/js/features/trip/components/Containers/CreateTripContainer.js]: use local state

File: resources/assets/js/features/trip/components/Containers/EditTripContainer.js [hunk 3]
```diff
@@ -38,10 +42,24 @@ class EditTripContainer extends React.Component {
                 place: null,
             }
         };
+
+        this.tripEndTime = 0;
```
Comment (TatsuUkraine) [inline resources/assets/js/features/trip/components/Containers/EditTripContainer.js]: use local state

File: resources/assets/js/features/trip/components/Containers/EditTripContainer.js [hunk 4]
```diff
@@ -38,10 +42,24 @@ class EditTripContainer extends React.Component {
...
+        this.waypointsDurations = [];
```
Comment (TatsuUkraine) [inline resources/assets/js/features/trip/components/Containers/EditTripContainer.js]: use local state

**Review by TatsuUkraine — CHANGES_REQUESTED** [review]

**Comment by TatsuUkraine** [comment]
@ruig1992 resolve conflics

**Comment by ruig1992** [comment]
@TatsuUkraine  Conflicts resolved.

**Review by sergey-ben — APPROVED** [review]

**Review by TatsuUkraine — APPROVED** [review]

**Pull request merged**